server/src/main/java/io/crate/replication/logical/action/PublicationsStateAction.java

```diff
@@ -163,13 +175,41 @@ protected ClusterBlockException checkBlock(Request request,
         }
 
         @VisibleForTesting
-        static List<RelationName> resolveRelationsNames(Publication publication, Schemas schemas) {
+        static User ensurePublicationOwnerExists(String publicationOwner, UserLookup userLookup, String publicationName) {
+            var user = userLookup.findUser(publicationOwner);
+            if (user == null) {
+                throw new IllegalStateException(
+                    String.format(Locale.ENGLISH, "User %s owning the publication %s is not found, stopping replication.",
+                        publicationOwner,
+                        publicationName
+                    ));
+            }
+            return user;
+        }
+
+        @VisibleForTesting
+        static User ensureSubscriberExists(String subscriber, UserLookup userLookup, String publicationName) {
+            var user = userLookup.findUser(subscriber);
+            if (user == null) {
+                throw new IllegalStateException(
+                    String.format(Locale.ENGLISH, "User %s subscribed to the publication %s is not found, stopping replication.",
+                        subscriber,
+                        publicationName
+                    ));
+            }
+            return user;
+        }
```

Comment:
both methods do the same apart from the message, couldn't we just use one impl. and pass in the concrete message?

Reply:
🤦 -  that's Friday afternoon PR)